### PR TITLE
[Op][Refactor] Decl make-functions in local headers. Namespace alignment cont.

### PR DESCRIPTION
**Pull request by MasterJH5574**
Following #81, the main focus of this PR is to expose the make-functions of every operators into local headers. And meanwhile remove the prepended `"Make"` prefix of the make-function names, so that whenever people want to call certain operator, they can make it by include the corresponding header and invoke the make-functions, without constructing the attribute and the op call manually.

Besides, this PR continues to align the name/namespace of the operators so far, according to [DataAPI](https://data-apis.org/array-api/latest/API_specification/index.html). Specifically, this PR
* move Matmul from NN namespace to “linear algebra” namespace;
* rename Variance to Var.

Several to-do items or undetermined items:
- [x] Do the make-function refactoring for the builtin operators, memory operators, etc.
- [x] Update the use sites of operator calls.
- [x] Rename Cast to Astype per DataAPI (?) https://data-apis.org/array-api/latest/API_specification/generated/array_api.astype.html#array_api.astype

cc @tqchen

### CODE REVIEW

**Comment by MasterJH5574** [comment]
Well it seems that changing “variance” to “var” will come with conflict, as “var” is usually understood as “variable.” I’ll change it back with proper notes.

File: src/relax/op/tensor/statistical.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  Expr OpName(Expr data, Optional<Array<Integer>> axis, bool keepdims) { \
+    ObjectPtr<StatisticalAttrs> attrs = make_object<StatisticalAttrs>(); \
+    attrs->axis = std::move(axis);                                       \
+    attrs->keepdims = keepdims;                                          \
+    static const Op& op = Op::Get("relax." OpRegName);                   \
+    return Call(op, {std::move(data)}, Attrs{attrs}, {});                \
+  }                                                                      \
+  TVM_REGISTER_GLOBAL("relax.op." OpRegName).set_body_typed(OpName)
+
+Expr Sum(Expr data, Optional<Array<Integer>> axis, bool keepdims);
```
Comment (tqchen) [inline src/relax/op/tensor/statistical.h]: naming, use low-level case Sum=>sum to be consistent with data api

File: src/relax/op/nn/nn.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#define TVM_RELAX_OP_NN_NN_H_
+
+#include <tvm/relax/attrs/nn.h>
+
+#include "../op_common.h"
+
+namespace tvm {
+namespace relax {
+
+Expr ReLU(Expr e);
```
Comment (tqchen) [inline src/relax/op/nn/nn.h]: add some docstring given it is in header

File: src/relax/op/tensor/statistical.h (hunk 1 above)
Comment (MasterJH5574) [reply src/relax/op/tensor/statistical.h]: If so then many operators need be renamed to lowercase. Will do later.

**Comment by MasterJH5574** [comment]
@tqchen Some notes on the latest commit:
* Renamed all op to lowercase on C++ side, including those in DataAPI and those not in, basically to keep consistency.
* Added docstring to the op decl in headers. Not too much docstring content for NN ops and image ops though. They are well documented on Python side.
* Enhanced the macros for binary and statistical ops.

**Review by tqchen — APPROVED** [review]

**Pull request merged**